test/unit/app/controllers/permissions/permissions-controller-test.js

```diff
@@ -0,0 +1,1392 @@
+        permController.pendingApprovalOrigins.has(ORIGINS.c),
+        'pending approvals origins has expected item',
+      )
+    })
+
+    it('should throw if adding origin twice', function () {
+
+      const id = nanoid()
+      const origin = ORIGINS.a
+
+      permController._addPendingApproval(id, origin, noop, noop)
+
+      const otherId = nanoid()
+
+      assert.throws(
+        () => permController._addPendingApproval(otherId, origin, noop, noop),
+        ERRORS.pendingApprovals.duplicateOriginOrId(otherId, origin)
+      )
+
+      assert.equal(
+        permController.pendingApprovals.size, 1,
+        'pending approvals has single entry',
+      )
+
+      assert.equal(
+        permController.pendingApprovalOrigins.size, 1,
+        'pending approvals origins has single item',
+      )
+
+      assert.deepEqual(
+        permController.pendingApprovals.get(id),
+        { origin, resolve: noop, reject: noop },
+        'pending approvals should have expected entry'
+      )
+
+      assert.ok(
+        permController.pendingApprovalOrigins.has(origin),
+        'pending approvals origins has expected item',
+      )
+    })
+
+    it('should throw if adding id twice', function () {
+
+      const id = nanoid()
+      const origin = ORIGINS.a
+
+      permController._addPendingApproval(id, origin, noop, noop)
+
+      assert.throws(
+        () => permController._addPendingApproval(id, ORIGINS.b, noop, noop),
+        ERRORS.pendingApprovals.duplicateOriginOrId(id, ORIGINS.b)
+      )
+
+      assert.equal(
+        permController.pendingApprovals.size, 1,
+        'pending approvals has single entry',
+      )
+
+      assert.equal(
+        permController.pendingApprovalOrigins.size, 1,
+        'pending approvals origins has single item',
+      )
+
+      assert.deepEqual(
+        permController.pendingApprovals.get(id),
+        { origin, resolve: noop, reject: noop },
+        'pending approvals should have expected entry'
+      )
+
+      assert.ok(
+        permController.pendingApprovalOrigins.has(origin),
+        'pending approvals origins has expected item',
+      )
+    })
+  })
+
+  describe('miscellanea and edge cases', function () {
+
+    let permController, notifications
+
+    beforeEach(function () {
+      notifications = initNotifications()
+      permController = initPermController(notifications)
+    })
+
+    it('notifyDomain handles notifications other than accountsChanged', async function () {
+
+      permController.notifyDomain(ORIGINS.a, NOTIFICATIONS.test())
+
+      assert.deepEqual(
+        notifications[ORIGINS.a][0],
+        NOTIFICATIONS.test(),
+        'origin should have correct notification'
+      )
+    })
+
+    it('addInternalMethodPrefix', function () {
+      const str = 'foo'
+      const res = addInternalMethodPrefix(str)
+      assert.equal(res, WALLET_PREFIX + str, 'should prefix correctly')
+    })
+
+    it('eth_accounts restricted method failure', async function () {
```

Comment:
It's unclear to me how this test relates to the permissions controller

Reply:
Well, we need to test the restricted method implementations, and there's only the one. Would be worthwhile to break out into its own file once we add more restricted methods, but for now it's fine?

Reply:
The way our unit tests are structured elsewhere is per-module; one set of tests around each one. Why break from that convention? That's how I find tests. If I were to go looking for tests of `restrictedMethods.js`, I'd assume there were none if I didn't find a test file named as such.

Reply:
Good point, fixed in 44ad320